### DISCUSSION TITLE
[논의사항] 규칙을 좀 더 정하는 것이 어떤가요?

### DISCUSSION THREAD

**Discussion by jhsseonn** [SSAFYSS/board-practice#7] [논의사항]
1주차 진행해보면서 SpringBoot + MyBatis 조합을 처음 써봐서 설정부터 많이 막혔습니다...
특히 롬복을 안 쓰니 불편한 부분이 상당히 많았네요.. 
그래서!!! 논의하고 싶었던 것들 제안드려봅니다.

1. lombok 사용
- `@Getter` `@Setter` `@NoArgsConstructor` `@Builder` 등... 사용하면 코드의 양이 대폭 줄어들고 편리한 어노테이션들이 많음
- 문제는 이 모든 어노테이션을 롬복이 제공함
- 처음에 싸피에서 배운 스프링이 레거시라 관통까지 레거시 방식을 따라가기로 논의했으나 이번주 작업하면서 굉장히 불편하다고 생각했음...ㅇ
- 결론: 롬복은 쓰는 것이 어떻습니까 JPA는 나중에 쓰더라도...
- 롬복을 사용하게 되면 논의할 사항 : 엔티티 생성시 생성자/빌더 선택, 어노테이션 사용(`@Setter` 지양, `@NoArgsConstructor` 기본생성자 사용 등) 규칙 정하기
- 롬복에 대해 알아보기 : [[Java] Lombok이란? 및 Lombok 활용법](https://mangkyu.tistory.com/78)

2. DTO명, 메서드명, 변수명 등 통일
- 이름 통일에 대해선 대충 지나가듯이 이야기 해서 이번주 작업한 것도 사람마다 매우 다를 것으로 예상됨
- 작업한 내용을 하나로 합치려면 통일 필요함

3. DTO - class? record?
- 객체 생성 시 생성자, getter, toString 등 반복되는 코드를 알아서 만들어주는 record라는 객체 타입이 있음! 한번 츄라이 해보실래요?
- DTO의 경우 단순 데이터 운반 역할로 사용할 때는 생성자 외에 값을 변경하는 Setter가 필요하지 않음 -> 불변 객체로 사용해도 무방함 -> 그렇다면 쓸데없이 코드가 길어지는 것을 방지하기 위한 방법으로 record를 사용하는 것도 나쁘지 않다!
- Record에 대해 알아보기 : [[Java] record에 대하여](https://velog.io/@pp8817/record)

이정도만 정해도 추후 합칠 때 문제가 덜할 것으로 생각됩니다. 혹시 추가 의견 있으면 주셔도 좋습니다!
@JJBINY 팀원은 아니지만 의견 대환영입니다

**Comment by JJBINY** [comment]
1. Lombok을 사용하는 것은 좋지만 말씀하신 것처럼 Setter 지양 등 컨벤션을 잘 만들어야 할 것 같아요.
만약 사용한다면 @Builder를 쓰는 것도 한 번 고려해보면 좋을 것 같아요. 자동으로 Builder 패턴을 적용해주는 애노테이션입니다.

3. 동의합니다. 따로 코드 컨벤션을 작성해 보는 게 어떨까요?

4. DTO에 Record를 사용하는 건 좋다고 생각해요. 불변이기 때문에 데이터 전달이라는 DTO 본연의 역할에 적합합니다.
메서드를 작성할 수 없기 때문에 누군가 실수로 DTO에 비즈니스 로직을 삽입하는 경우를 어느 정도 막아줄 수도 있어요.

**Comment by beegle98** [comment]
1. lombok은 쓰는게 좋을 것 같아요!
2. 통일의 필요성을 뼈저리게 느끼고 있어요
3. 찬성찬성!

**Comment by JJBINY** [comment]
PR 작성할 때 전부는 아니더라도 중요한 로직 정도는 왜 이런 방식 혹은 구조로 작성했는지, 어떤 부분들을 고민했는지 3~5줄 정도로 간단히 코멘트 남겨주시면 좋을 거 같아요!
작성한 로직을 쉽게 파악 및 비교할 수 있고, 생각을 정리하면서 잘못된 부분이나 보완해야 할 점을 깨닫는 데 도움이 될 것 같아요.
